go/runtime/bundle/manager.go

```diff
@@ -236,20 +246,46 @@ func (m *Manager) Queue(runtimeID common.Namespace, manifestHashes []hash.Hash)
 	}
 	m.downloadQueue[runtimeID] = hashes
 
-	// Trigger immediate download of new bundles.
+	// Trigger immediate download and clean-up of bundles.
+	select {
+	case m.triggerCh <- struct{}{}:
+	default:
+	}
+}
+
+// Cleanup updates the runtime's maximum bundle version for pending clean-up.
+//
+// Existing maximum version in the clean-up queue for the specified runtime (if any),
+// is removed and replaced with the given one.
+//
+// Warning: If clean-up fails it's not retried.
+func (m *Manager) Cleanup(runtimeID common.Namespace, version version.Version) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	m.cleanupQueue[runtimeID] = version
```

Comment:
```suggestion
	m.cleanupQueue[runtimeID] = max(version, m.cleanupQueue[runtimeID])
```
This is not the best. If I request cleanup for version 2, I expect it to execute, even if I request cleanup for version 1 afterwards.

Reply:
Hm technically `Cleanup` docstring does not promise you this? Agree is useful and can update also the docstring then...? On the side note don't we have the same problem with `Download` then?

Reply:
It doesn't promise, but it should. If my code sees stale bundles, and calls clean to remove them, I expect them to be removed as they existed at that time. I'm not happy if someone can cancel that.

Yes, `Download` cancels previous versions, because that is useful. It is not useful in cleanups. We could refactor `Download` to not cancel queued versions, but then you have to:
- cancel them on your own, e.g. by looking at max version from cleanup, or,
- add `CancelDownloads` method.

The latter solution would be nice, and easy to implement. But probably the work will be wasted once we move version listening to the manager.

Reply:
Makes sense. I updated also the docstring.

`max` does not exist since version is not ordered type. I have replaced your code with:


```go
if maxVersion, ok := m.cleanupQueue[runtimeID]; !ok || maxVersion.Less(version) {
	m.cleanupQueue[runtimeID] = version
}
```

Technically this could be simplified with: 

```go
if m.cleanupQueue[runtimeID].Less(version) {
    m.cleanupQueue[runtimeID] = version
}
```

since we store version values not pointers in the queue, but feels a bit hackish to rely on a zero value.

What do you prefer?